src/test/java/de/tum/in/www1/artemis/architecture/ArchitectureTest.java

```diff
@@ -246,6 +249,27 @@ void ensureSpringComponentsAreProfileAnnotated() {
         rule.check(productionClasses);
     }
 
+    @Test
+    void dtosWithCollectionsHaveJsonAnnotation() {
+        var DTOs = allClasses.that(nameContaining("DTO"));
+        var DTOsWithCollections = DTOs.that(haveACollectionField());
+
+        ArchRule rule = classes().should().beAnnotatedWith(JsonInclude.class).because("empty collections should not be part of the json");
+
+        rule.check(DTOsWithCollections);
+    }
+
+    private DescribedPredicate<JavaClass> haveACollectionField() {
+        return new DescribedPredicate<>("have a field that is a collection") {
+
+            @Override
+            public boolean test(JavaClass javaClass) {
+                return javaClass.getAllFields().stream().map(field -> field.getRawType().reflect())
+                        .anyMatch(field -> field.equals(List.class) || field.equals(Set.class) || field.equals(Map.class) || field.equals(Collection.class));
```

Comment:
Can you here check for any subclasses of `Collection`?

Reply:
I already thought there has to be a simpler version, but I did not think of that...
Thank you!